ouroboros-consensus/test-consensus/Test/Dynamic/Util.hs

```diff
@@ -233,6 +234,13 @@ consensusExpected k nodeJoinPlan schedule =
   where
     NumBlocks maxForkLength = determineForkLength k nodeJoinPlan schedule
 
+emptyLeaderSchedule :: NumSlots -> LeaderSchedule
+emptyLeaderSchedule (NumSlots t) = LeaderSchedule $
```

Comment:
Is there an invariant somewhere that this satisfies (which `LeaderSchedule Map.empty` did not)? If so, then it's worth documenting it.

Reply:
I haven't seen such an invariant listed, no. The code seems to be in limbo about it: the invariant is apparently respected everywhere but it is not expected/required anywhere. The `Map.empty` I had here for a moment was the first time I've seen "missing" slots in a leader schedule, so I replaced it. It seemed reasonable to not spoil an apparent invariant without cause.

What do you think - go ahead an introduce the invariant and adjust the type's interface (in a separate PR)?

Reply:
I trust your judgment :slightly_smiling_face: